### PR TITLE
Dev/17 make reusable page layout

### LINKED ISSUE #17: make-reusable-page-layout
`Header.js` and `Footer.js` components for page layout at `\layouts`

### PR DESCRIPTION
**Pull request by mar-bi**
close #17

### CODE REVIEW

File: src/layouts/components/Footer.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import Link from 'gatsby-link';
+import styled from 'styled-components';
+
+import { ColorPalette as CP } from "../../theme/globalStyle";
+import PageNav from "./PageNav.js";
+
+const StyledFooter = styled.footer`
+  background: ${CP.primary.dark};
+  grid-area: ft;
```
Comment (spences10) [inline src/layouts/components/Footer.js]: what id `ft`? curious, not seen it before

File: src/layouts/components/Footer.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  grid-area: team;
+  color: ${CP.white};
+  padding: 0.5rem;
+  margin: 1rem;
+  font-size: 1rem;
+`;
+
+const Footer = () => (
+  <StyledFooter>
+    <PageNav header={false}/>
```
Comment (spences10) [inline src/layouts/components/Footer.js]: What does `header={false}` do?

File: src/layouts/components/Header.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import { StyledH1, ColorPalette as CP } from "../../theme/globalStyle";
+import PageNav from "./PageNav.js";
+
+const StyledHeader = styled.div`
+  grid-area: hd;
+  display: grid;
+  grid-template-columns: repeat(12, 1fr);
+  grid-template-rows: auto;
+  grid-template-areas:
+    ". brand brand . . . nav nav nav nav nav .";
```
Comment (spences10) [inline src/layouts/components/Header.js]: what is this `. brand brand . . . nav nav nav nav nav .` 🙃

**Comment by mar-bi** [comment]
Well, I've got great questions. :smiley: 

> what is this . brand brand . . . nav nav nav nav nav . :upside_down_face:

This is a very cool and easy way to create a `grid` on a `container`. 
I declared 12-cols grid and referenced the names of grid areas. 
A period `.` (with following by *space*) means  **an empty column** and `brand` or `ft` or `nav` or `main` means **named grid area**.
Here I declare how I plan to use these 12 cols:
``` 
grid-template-areas:
   ". brand brand . . . nav nav nav nav nav ." 
// empty col, 2 cols brand, 3 empty cols, 5 cols nav, empty col

  ". main main main main main main main main main main ." 
// empty col, 10 cols of main, empty col
  "ft ft ft ft ft ft ft ft ft ft ft ft" // 12 cols ft
```
Then, I can tell the `child` of the `container` how many cols it can occupy.
```
const StyledFooter = styled.footer`
  background: ${CP.primary.dark};
  grid-area: ft;  
// "Dear child, please take all the area i requested for ft". It will take all 12 cols.
```
I hope I make it clear. Actually, I find this way of building grid today :grinning: 
Here you can find more: 
https://css-tricks.com/snippets/css/complete-guide-grid/
https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout

**Comment by mar-bi** [comment]
> What does header={false} do?

When I started building `Header` and `Footer` I noticed both of them used the same `nav`.
I made `PageNav` as a separate component and add a conditional styling. 

If the `Header` is calling it, it sends a `header={true}` prop. The `Footer` sends `false`.
Then the `PageNav` sends this prop down to children which have the conditional styling.

Here is more: https://www.styled-components.com/docs/basics#adapting-based-on-props

P.S. I try to use 2 slightly different ways of this styling  for `StyledLi` and for `Link`.
(this one for `Link` https://www.styled-components.com/docs/basics#attaching-additional-props).
Both of them work. :smile:

**Comment by spences10** [comment]
Awesome! Thanks!!

**Pull request merged**